ssl/statem/statem_clnt.c

```diff
@@ -1886,10 +1886,6 @@ WORK_STATE tls_post_process_server_certificate(SSL *s, WORK_STATE wst)
         return WORK_ERROR;
     }
     ERR_clear_error();          /* but we keep s->verify_result */
-    if (i > 1) {
-        SSLfatal(s, SSL_AD_HANDSHAKE_FAILURE, i);
-        return WORK_ERROR;
-    }
```

Comment:
@mattcaswell Any objections?  I suggested to @DDvO that this is an obsolete check (see earlier comments in this PR)

Reply:
ping @mattcaswell

Reply:
It seems to have originated in b362ccab5c1 where, as Viktor discussed upthread, it could return the `ssl_security_cert_chain()` return value that would in turn be able to match this check.
However, in the current incarnation, `i` can still be the return value of an `app_verify_callback` (or `X509_verify_cert()` if no callback is specified).  While the current documentation for SSL_CTX_set_cert_verify_callback() says that the callback should return only 1 or 0 (or -1 for client certificates), I would not be surprised if some application somewhere has come to rely upon this (undocumented) behavior (which would have to be in the 7 years since the check was introduced, not in the 19 years since the callback documentation was introduced).  As such, I'm reluctant to support the change (but I also am not actively objecting to it).  The documentation appeared in 2001 in c4068186acec0f5cd370f7b5772b41f215903234 ("as discussed recently on openssl-users") but I'm not sure that I could find the email thread in question (or whether it would shed any additional light on the subject).

Reply:
@kaduk, I agree that you raise a valid concern, but I don't think that it should stop us from making the code simpler, more uniform.  We can do this in a major release (3.0), and have a clear notice in the NEWS and CHANGES files.  I don't know anywhere else in OpenSSL where <= 0 is an error, 1 is OK, but all other positive values are also failures.  This particular interface is unexpected behaviour, and I think we should change it to behave like the rest of the library.  The documentation does not promise the current semantics, so I think we're on pretty safe ground here...

@mattcaswell your thoughts?

Reply:
I agree that if we're going to do it, the procedure you describe is the right way to do it.
I'm not going to stand in the way of doing it, but I don't currently expect to add my own +1 to the change -- Matt should be able to handle that, though :)

Reply:
> and have a clear notice in the NEWS and CHANGES files.

I don't think that this detail would be worth a NEWS entry, but tentatively added to `CHANGES.md`:
```
 * While a callback function set via `SSL_CTX_set_cert_verify_callback()`
   is not allowed to return a value > 1, this is no more taken as failure.

   *Viktor Dukhovni and David von Oheimb*
   ```